### ISSUE TITLE
Chat coordinates

### ISSUE THREAD

**Issue by Nukecraft5419** [mrbuilder1961/WheresMyChatHistory#84]
**Is your feature request related to a problem? Please describe.**
Automatically sending current co-ordinates in chat at the push of a button

**Describe the solution you'd like**
Send co-ordinates in chat

**Describe alternatives you've considered**
https://modrinth.com/mod/chat-coords

**Additional context**
...

**Comment by mrbuilder1961** [comment]
ok i still haven't started this yet bc i know it's not going to be very hard and i want to save some easy impls for later, so sorry for the long wait. 

but i was thinking, is the option to send current AND nether coords a good idea? maybe like a little switch right next to the button or smth. i know i would use the hell out of that feature lol.

**Comment by Nukecraft5419** [comment]
Actually ChatPaches is in the LifeMC modpack of my team LifeMC Studios and I look for mods that have a lot of potential and I help them with ideas in addition to putting them in the modpack and sometimes I help through PR to update and improve the project, it's a great idea even if it's convenient to press a button even if as an alternative it's better to have a dedicated button one for overworld one for the nether and the other for the end or one that does everything I would take a cue from No Chat Reports for the button as a location

**Comment by EGOIST1372** [comment]
I dont think sendeing all the dimensions coord all in once be a good idea, end coord is useless. i only can think of nether coord with overworld coord just for linking nether portals thats it. so it can be done with ONLY one button. left click sends only current location. it just need to have a stamp that says overworld or nether or end as the current location(u might be in different dimensions so having that stamp is useful)
and for the overworld-nether combo coord right click on button

**Comment by Nukecraft5419** [comment]
It would also be cool if you already have a beta I can test it no problem

**Comment by mrbuilder1961** [comment]
in response to EGOIST: for the most part that all sounds good, could you give an example of what the stamp thing might look like?

in response to Nukecraft: no, i haven't started on this yet. i will provide a beta file at some point in the discord!

**Comment by Nukecraft5419** [comment]
I will wait for the beta file so at least I can help

**Comment by EGOIST1372** [comment]
Stamp like [Overworld] with green color
And [Nether] with red color
And [End] with purple
Something like this after time stamp part of the nessage

**Comment by mrbuilder1961** [comment]
okay i'd say this is needed pretty badly. should it be a button or command?

**Comment by Nukecraft5419** [comment]
> okay i'd say this is needed pretty badly. should it be a button or command?

It would be much better if as soon as you open the chat a dynamic button, for example when you are in the overworld an icon of that world appears, then in the nether the portal of the nether is in the end the portal of the end is pressed sends the coordinates to the chat.

**Comment by EGOIST1372** [comment]
> Stamp like [Overworld] with green color
> And [Nether] with red color
> And [End] with purple
> Something like this after time stamp part of the nessage
> […](#)



> I dont think sendeing all the dimensions coord all in once be a good idea, end coord is useless. i only can think of nether coord with overworld coord just for linking nether portals thats it. so it can be done with ONLY one button. left click sends only current location. it just need to have a stamp that says overworld or nether or end as the current location(u might be in different dimensions so having that stamp is useful) and for the overworld-nether combo coord right click on button

As I suggested before and I still stand by those suggestions
one thing that can be done and be useful is adding some abilities on [calcmod](https://modrinth.com/plugin/calcmod) into chat patches like nether to overworld and vice versa coords conversion
/calc nether ~ ~ ~
<img width="1280" height="87" alt="Image" src="https://github.com/user-attachments/assets/23cbba4d-b4ff-4f6e-be37-89fe0952a550" />

Portal Linking Coords

When given a block position, returns the dimension in the command's corresponding coordinates. If no coordinates are given, command assumes current player position.

    Usage: /calc nether <x> <y> <z>
    Usage: /calc overworld <x> <y> <z>

in my opinion showing normal coords on a chat mod is not useful since there are many many better solutions that can be used for a coords display on the hud
so I only suggesting adding abilities on chat patches that be useful like nether linking stuff and even more features from calcmod

**Comment by EGOIST1372** [comment]
> okay i'd say this is needed pretty badly. should it be a button or command?

@mrbuilder1961 
But if you want to add this feature really badly I would say a button that says Coords and when you press it it send nether and overworld correspond coords with the stamps i mentioned in my previous comments (end coords are useless)

**Comment by EGOIST1372** [comment]
Another idea that I say would be useful is pin a message
it can be extended to a waypoint/checkpoint/pinpoint feature as well (related to current discussion on coords chat)
let's say I go and find a woodland mansion or a beautiful place or mob spawner and I type the coords on my chat to remember it next time and to not lose it I want to pin it so it will be at my sight all the time

now features that I imagine for this idea are:

1. when right click on a message with mouse a drop down menu opens (already exists) and option to pin and unpin is there when hovered on pin, another drop side menu opens with 2 options
2. option 1, pin to world chat (only pins to that world's chat and always will be shown when chat is hidden and its position will be one message above the armor bar so there will be no visual interference)
3. option 2, pin to global chat (pins to all worlds and servers chat and always will be shown when chat is hidden and its position will be one message above the armor bar so there will be no visual interference)
4. you can pin multiple messages one by one and they stack on each other
5. unpin option only will be shown on pinned messages when right clicking on them (if this will be hard to implement then just show unpin option all the time for all messages, users will understand how to use it on pinned messages)
6. color coordinate messages (when clicking on option 1 or option 2 in drop side menu there will be another drop side menu that shows 6 colors to choose from to color coordinate my messages and I'm thinking on red, green,blue,purple,gold/yellow and black bold)

**Comment by Aces-and-Jacks** [comment]
> As I suggested before and I still stand by those suggestions one thing that can be done and be useful is adding some abilities on [calcmod](https://modrinth.com/plugin/calcmod) into chat patches like nether to overworld and vice versa coords conversion /calc nether ~ ~ ~ <img alt="Image" width="1280" height="87" src="https://private-user-images.githubusercontent.com/86555131/471462887-23cbba4d-b4ff-4f6e-be37-89fe0952a550.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njk1MTIwNDUsIm5iZiI6MTc2OTUxMTc0NSwicGF0aCI6Ii84NjU1NTEzMS80NzE0NjI4ODctMjNjYmJhNGQtYjRmZi00ZjZlLWJlMzctODlmZTA5NTJhNTUwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAxMjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMTI3VDExMDIyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA2MWRlZjI1M2NlZjM4Yjc1ZDgyM2FkYmFjZmU1NzM3Y2EzMDQyNDcwZDY0ODAxYWQ4ZjIzMGU5NDU5N2ZiZWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.dxgWd_nuhRcup8rE-rQ-2rnDohnYHpp4XS_7bMy4Qzo">
> 
> Portal Linking Coords
> 
> When given a block position, returns the dimension in the command's corresponding coordinates. If no coordinates are given, command assumes current player position.
> 
> ```
> Usage: /calc nether <x> <y> <z>
> Usage: /calc overworld <x> <y> <z>
> ```
> 
> in my opinion showing normal coords on a chat mod is not useful since there are many many better solutions that can be used for a coords display on the hud so I only suggesting adding abilities on chat patches that be useful like nether linking stuff and even more features from calcmod

Calcmod is still being kept up to date. And it already does what it does well. I don't really think coordinates should be included in this mod if a mod like that exists. A player can just copy the message and send it in chat. If coordinates is going to be implemented then I think it should just be current coordinates and not other dimensions as well

**Comment by mrbuilder1961** [comment]
unfortunately i'm gonna have to agree with Aces here. i apologize for going back on this issue especially because it's been open for so long but i really don't see  how a chat coordinate button fits in with all the existing chat features. 

it feels like its own mod, and i don't want to reimplement another mod that is up to date and working well already, that's just a recipe for burnout and disappointment.

again, sorry for dragging this out, but the current vision for chat patches doesn't align with a coordinate button. id recommend this calc mod mentioned (although i haven't tested it myself 🌚)!

**Comment by Nukecraft5419** [comment]
> unfortunately i'm gonna have to agree with Aces here. i apologize for going back on this issue especially because it's been open for so long but i really don't see how a chat coordinate button fits in with all the existing chat features.
> 
> it feels like its own mod, and i don't want to reimplement another mod that is up to date and working well already, that's just a recipe for burnout and disappointment.
> 
> again, sorry for dragging this out, but the current vision for chat patches doesn't align with a coordinate button. id recommend this calc mod mentioned (although i haven't tested it myself 🌚)!

Yes, I agree with both of you. If it's complicated to add that feature, but there's already a mod that does it, then it's better to add it to the modpack than to burn out.
